app/controllers/api/v2/job_invocations_controller.rb

```diff
@@ -18,7 +18,20 @@ def index
 
       api :GET, '/job_invocations/:id', N_('Show job invocation')
       param :id, :identifier, :required => true
+      param :host_status, String, required: false, allow_blank: true, desc: N_('Show Job status for the hosts.')
       def show
+        @hosts = @job_invocation.targeting.hosts.authorized(:view_hosts, Host)
+        @template_invocations = @job_invocation.template_invocations
+                                               .where(host: @hosts)
+                                               .includes(:input_values)
+
+        if params[:host_status]
+          @template_invocations.includes(:run_host_job_task)
+          @hosts.map do |host|
```

Comment:
Map iterates over elements in the enumerable, applying the block to each of the elements and collecting the results into an array. In here it would generate an array of template invocation statuses, which would be immediatelly garbage collected since they're not being used anywhere. `#each` would be a better fit in this case since your modifying already existing objects

Reply:
No idea why I used `map` instead of `each` ...